src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs

```diff
@@ -118,6 +119,11 @@ public override string ToString()
 
             return m_toString;
         }
+
+        public override bool Equals(object? obj) =>
+            obj is RuntimeConstructorInfo ci && m_handle == ci.m_handle && ReferenceEquals(m_declaringType, ci.m_declaringType);
```

Comment:
Probably should have a `ReferenceEquals(this, obj) ||` at the beginning.

Reply:
Not sure if that would help with perf, need to test

Reply:
I think that the perf of the two options is close enough. There is a value in simplicity too.

Reply:
> I think that the perf of the two options is close enough

Yes there were no noticeable difference, but GetHashCode was quite slow, so added `MetadataUpdater.IsSupported` to conditionally use the old approach `base.GetHashCode()`